### PR TITLE
do not change speed of playing decks if "Reset Speed/Tempo" preference is checked

### PR DESCRIPTION
**Pull request by Be-ing**
Fixes https://bugs.launchpad.net/mixxx/+bug/1603591

I'll write a test to prevent a regression, so don't merge this yet.

### CODE REVIEW

**Comment by Be-ing** [comment]
Okay, test written and ready for review. I had to do some refactoring of MockedEngineBackendTest to do so. All the EngineSyncTest tests are still working. I can't confirm the refactoring didn't break any other tests because other tests are crashing on my system:
https://bugs.launchpad.net/mixxx/+bug/1620322

File: src/engine/sync/enginesync.cpp [hunk 1]
```diff
@@ -397,3 +397,19 @@ EngineChannel* EngineSync::pickNonSyncSyncTarget(EngineChannel* pDontPick) const
     // had a BPM.
     return pFirstNonplayingDeck;
 }
+
+bool EngineSync::otherSyncedPlaying(const QString& group) {
+    int othersInSync = 0;
+    foreach (Syncable* theSyncable, m_syncables) {
```
Comment (daschuer) [inline src/engine/sync/enginesync.cpp]: use the new C++11 range based loop
```for (Syncable* theSyncable: m_syncables)```

File: src/engine/sync/enginesync.cpp [hunk 2]
```diff
@@ -397,3 +397,19 @@ EngineChannel* EngineSync::pickNonSyncSyncTarget(EngineChannel* pDontPick) const
...
+        if (theSyncable->getGroup() == group) {
+            if (theSyncable->getSyncMode() == SYNC_NONE) {
+                return false;
+            }
+            continue;
+        }
+        if (theSyncable->isPlaying() && (theSyncable->getSyncMode() != SYNC_NONE)) {
+            othersInSync++;
```
Comment (daschuer) [inline src/engine/sync/enginesync.cpp]: It looks like you can return true here.

File: src/mixer/basetrackplayer.cpp [hunk 3]
```diff
@@ -289,24 +291,19 @@ void BaseTrackPlayerImpl::slotTrackLoaded(TrackPointer pNewTrack,
...
+            // are playing, as this would change the speed of already playing decks.
+            if (! m_pEngineMaster->getEngineSync()->otherSyncedPlaying(getGroup())) {
+                if (m_pRateSlider != NULL) {
+                    m_pRateSlider->set(0.0);
+                }
             }
-            M_FALLTHROUGH_INTENDED;
-          case RESET_PITCH:
+        }
+        if (reset == RESET_PITCH || reset == RESET_PITCH_AND_SPEED) {
```
Comment (daschuer) [inline src/mixer/basetrackplayer.cpp]: else if

File: src/engine/sync/enginesync.cpp (hunk 2 above)
Comment (daschuer) [inline src/engine/sync/enginesync.cpp]: No, sorry, I am wrong.  
But why not make othersInSync bool?

File: src/mixer/basetrackplayer.cpp (hunk 3 above)
Comment (Be-ing) [inline src/mixer/basetrackplayer.cpp]: No, this should be a separate `if` statement. `else if` would require a third branch with duplicate code for RESET_PITCH_AND_SPEED

File: src/engine/sync/enginesync.cpp [hunk 4]
```diff
@@ -397,3 +397,19 @@ EngineChannel* EngineSync::pickNonSyncSyncTarget(EngineChannel* pDontPick) const
     // had a BPM.
     return pFirstNonplayingDeck;
 }
+
+bool EngineSync::otherSyncedPlaying(const QString& group) {
```
Comment (Be-ing) [inline src/engine/sync/enginesync.cpp]: If you want to validate the test, insert `return false;` here.

File: src/engine/sync/enginesync.cpp (hunk 2 above)
Comment (Be-ing) [inline src/engine/sync/enginesync.cpp]: done

File: src/engine/sync/enginesync.cpp (hunk 1 above)
Comment (Be-ing) [inline src/engine/sync/enginesync.cpp]: done

**Comment by daschuer** [comment]
Travis fails, not your fault: 
The job exceeded the maximum time limit for jobs, and has been terminated.

**Comment by daschuer** [comment]
LGTM Thank you!

**Comment by Be-ing** [comment]
Good catch. Yes, I'll take a closer look at this.

**Pull request merged**